app/experimenter/static/js/components/Pref.js

```diff
@@ -0,0 +1,143 @@
+import { boundClass } from "autobind-decorator";
+import { Map } from "immutable";
+import PropTypes from "prop-types";
+import React from "react";
+import { Button, Row, Col } from "react-bootstrap";
+import DesignInput from "experimenter/components/DesignInput";
+
+import {
+  PREF_KEY_HELP,
+  PREF_TYPE_HELP,
+  PREF_BRANCH_HELP,
+} from "experimenter/components/constants";
+import { PREF_VALUE_HELP } from "./constants";
+
+@boundClass
+class Pref extends React.PureComponent {
+  static propTypes = {
+    preference: PropTypes.instanceOf(Map),
+    errors: PropTypes.instanceOf(Map),
+    index: PropTypes.number,
+    onChange: PropTypes.func,
+    remove: PropTypes.func,
+  };
+
+  handleChange(key, value) {
```

Comment:
Remember the pattern I was talking about, `handleChange` should include WHAT's changing ie `handleWhateverChange`